accounts-db/src/accounts_db.rs

```diff
@@ -8026,6 +8034,7 @@ impl AccountsDb {
                     dead_slots.insert(*slot);
                 }
                 else {
+                    let (_, us) = measure_us!({
```

Comment:
this code block could be sent to an async thread along with an `Arc` of the storage for the slot. The bg thread could work through these size lookups and update the global to shrink list when it is done.

like this:
https://github.com/anza-xyz/agave/blob/ff87ed9187ab5e4967ad6cfa61bb951fe54fb68b/accounts-db/src/accounts_db.rs#L2906-L2915